src/algebra/ordered_monoid.lean

```diff
@@ -396,6 +396,51 @@ instance [add_comm_monoid α] : add_comm_monoid (with_top α) :=
   ..@additive.add_comm_monoid _ $ @comm_monoid_with_zero.to_comm_monoid _ $
     @with_zero.comm_monoid_with_zero (multiplicative α) _ }
 
+section sub_neg
+
+@[priority 100]  -- see Note [lower instance priority]
+instance [has_neg α] : has_neg (with_top α) :=
+⟨option.map (λ (a : α), -a)⟩
+
+@[simp] lemma neg_top [has_neg α] : - (⊤ : with_top α) = ⊤ := rfl
+lemma neg_coe [has_neg α] (x : α) : (-x : with_top α) = ((-x : α) : with_top α) := rfl
```

Comment:
Is there a reason you have this backwards WRT the add and sub versions?

Reply:
I'll make it more akin to `coe_add`. I've noticed that due to the way coercions work, I'm almost always rewriting with `←coe_add` and `←coe_sub`. Really, all of these should be tagged with `[norm_cast]` or however so that they can be correctly `norm_cast` and `push_cast`ed.